server/src/main/java/org/opensearch/transport/InboundHandler.java

```diff
@@ -184,79 +192,83 @@ private <T extends TransportRequest> void handleRequest(TcpChannel channel, Head
         final String action = header.getActionName();
```

Comment:
@Gaganjuneja One important thing that is missed in this pull request is context propagation for transport action through headers (as for HTTP actions).

Reply:
Header is already been set in the ThreadContext so framework should should automatically be able to use it for parentSpan creation.

Reply:
https://github.com/opensearch-project/OpenSearch/blob/8807d7a801898c416dfefa532bb075a8edde0d5c/server/src/main/java/org/opensearch/transport/InboundHandler.java#L130

Reply:
Thanks @Gaganjuneja